ui/app/AppLayouts/Chat/controls/community/CommunityPermissionsRow.qml

```diff
@@ -0,0 +1,315 @@
+                   (holdingsCount1 === 2 && holdingsCount2 === 1 && holdingsCount3 === 1) ||
+                   (holdingsCount1 === 1 && holdingsCount2 === 2 && holdingsCount3 === 1) ||
+                   (holdingsCount1 === 1 && holdingsCount2 === 1 && holdingsCount3 === 2)) {
+                    shortModel.append(permission1)
+                    shortModel.append(permission2)
+                    shortModel.append(permission3)
+                    return
+                }
+            }
+
+            // CASE 3: More than 2 permissions and didn't fit with previous conditions (dots visualized)
+            if(modelCount > maxVisualPermissions) {
+                dotsVisible = true
+                shortModel.append(permission1)
+
+                // More than 2 permissions but 1st and 2nd have only 1:1 or 1:2 tokens
+                if((holdingsCount1 === 1 && holdingsCount2 === 1) ||
+                   (holdingsCount1 === 2 && holdingsCount2 === 1) ||
+                   (holdingsCount1 === 1 && holdingsCount2 === 2)) {
+                    shortModel.append(permission2)
+                }
+            }
+        }
+    }
+
+    implicitHeight: 24
+    spacing: 4
+    padding: 1
+
+    background: Rectangle {
+        color: root.backgroundColor
+        radius: root.backgroundRadius
+    }
+
+    contentItem: RowLayout {
+        id: container
+
+        anchors.centerIn: parent
+        anchors.margins: root.padding
+        spacing: root.spacing
+
+        StatusIcon {
+            Layout.preferredHeight: container.height - 6
+            Layout.preferredWidth: Layout.preferredHeight
+            Layout.leftMargin: 4
+
+            icon: root.requirementsMet ? "tiny/unlocked" : "tiny/locked"
+            color: Theme.palette.baseColor1
+        }
+
+        Repeater {
+            id: repeater
+
+            model: shortModel
+
+            RowLayout {
+                spacing: root.spacing
+
+                SinglePermissionRow {
+                    model: holdingsListModel
+                }
+
+                StatusBaseText {
+                    visible: index !== (repeater.count - 1) || d.dotsVisible
+                    text: qsTr("or")
+                    font.pixelSize: root.pixelSize
+                    color: Theme.palette.baseColor1
+                }
+            }
+        }
+
+        StatusRoundedComponent {
+            Layout.preferredHeight: container.height
+            Layout.preferredWidth: Layout.preferredHeight
+
+            visible: d.dotsVisible
+            color: Theme.palette.baseColor3
+            border.color: root.backgroundColor
+            border.width: root.overlappingBorder
+
+            StatusIcon {
+                anchors.centerIn: parent
+                visible: d.dotsVisible
+                icon: "dots-icon"
+                height: root.dotsIconSize
+                width: height
+            }
+        }
+    }
+
+    onModelChanged: d.buildShortModel(root.model)
```

Comment:
It will be triggered only when model is re-assigned. When model is changing internally, it won't be called. So if permissions change for given community, the permissions row won't be updated. Not sure how big problem it is.